### DISCUSSION TITLE
Pre-request Script doesn't run when using Flow

### DISCUSSION THREAD

**Discussion by claytoncc** [postmanlabs/postman-flows#332] [Defect]
Help!

I use the **Pre-request Script** to generate random data and set some **environment variables**. When I run a **Flow** the **Pre-request Script** is never called and **environment variables** are never set! The same issue for **Request Tests**, I use them to set **environment variables** as well, they are never called when using **Flows**

Can anyone fix this?

**Comment by samay-v** [comment]
Hey @claytoncc 
the Pre-request Script do work in flows. but the ENV and Global variables are treated as readonly. a detailed rational and workarounds was written out here https://github.com/postmanlabs/postman-flows/discussions/142
